framework/math/ParallelVector/ParallelSTLVector.cc

```diff
@@ -79,12 +82,84 @@ double& ParallelSTLVector::operator[](const int64_t local_id)
   return values_[local_id];
 }
 
+void ParallelSTLVector::Set(const double value)
+{
+  values_.assign(values_.size(), value);
+}
+
 void ParallelSTLVector::Set(const std::vector<double>& local_vector)
 {
-  ChiInvalidArgumentIf(local_vector.size() != local_size_,
+  ChiInvalidArgumentIf(local_vector.size() < local_size_,
                        "Incompatible local vector size.");
 
-  values_ = local_vector;
+  // We cannot assign values = local_vector because this might
+  // destroy the internals of a ghosted vector
+  for (size_t i = 0; i < local_size_; ++i)
+    values_[i] = local_vector[i];
```

Comment:
Since we are doing it elsewhere we could use `std::copy` here... but I will leave this up to you. I am fine either way. Just depends on whether we want to bulk do that everywhere we can or not.

Reply:
Done